catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3SessionsManager.java

```diff
@@ -224,24 +229,7 @@ Credentials sessionCredentialsForClient(String repositoryId, S3BucketOptions opt
   private static SessionKey buildSessionKey(String repositoryId, S3BucketOptions options) {
     // Client parameters are part of the credential's key because clients in different regions may
     // issue different credentials.
-    return ImmutableSessionKey.builder()
-        .repositoryId(repositoryId)
-        .region(
```

Comment:
We need so many options that I figured it was easier to store the whole `S3BucketOptions` in the key. @dimas-b do you think that's ok in terms of cache key lookup efficiency?

Reply:
Just make sure that `hashCode` is not computed every time

Reply:
`SessionKey` is annotated with `@NessieImmutable` so I think we are good.